lib/shrine/attachment.rb

```diff
@@ -34,30 +34,28 @@ def initialize(name, **options)
       # Defines attachment methods for the specified attachment name. These
       # methods will be added to any model that includes this module.
       def define_attachment_methods!
-        attachment    = self
-        attacher_ivar = :"@#{@name}_attacher"
+        attachment = self
+        name = attachment_name
 
-        define_method :"#{@name}_attacher" do |options = {}|
-          if !instance_variable_get(attacher_ivar) || options.any?
-            instance_variable_set(attacher_ivar, attachment.build_attacher(self, options))
+        define_method "#{name}_attacher" do |reload = false, **options|
+          if reload || options.any? || !instance_variable_get("@#{name}_attacher")
+            instance_variable_set("@#{name}_attacher", attachment.build_attacher(self, options))
           else
-            instance_variable_get(attacher_ivar)
+            instance_variable_get("@#{name}_attacher")
```

Comment:
Here as well, I would prefer to convert the instance variable name to a symbol `:"@#{name}_attacher"` in this method.